runtime/src/snapshot_utils.rs

```diff
@@ -1417,15 +1416,14 @@ pub fn bank_fields_from_snapshot_archives(
         full_snapshot_archive_info.slot(),
     );
 
-    let temp_dir = tempfile::Builder::new()
-        .prefix("dummy-accounts-path")
-        .tempdir()?;
+    let temp_unpack_dir = TempDir::new()?;
+    let temp_accounts_dir = TempDir::new()?;
```

Comment:
Agreed on the general cleanup.

One item that I'll note is that I don't love the use of temporary directories without a base path specified. At least once, I have hit (and I have seen operators hit it too) a failure to write to disk because `/tmp/` was filled up. This type of issue is a little tricky to identify given that operator may not think to check `/tmp/`.

The way that `/tmp/` could get filled up is if the process dies before the temporary directory goes out of scope and auto-deletes the contents. For example, if any of the calls after creating the temp directory panic.

This is arguably out of scope for this review, so not saying we need to address right now, but just came to mind seeing the temp directory use in the diff.